lib/cli.js

```diff
@@ -0,0 +1,54 @@
+"use strict";
+
+let { abort, repr } = require("../util");
+let parseArgs = require("minimist");
+
+let HELP = `
+Usage:
+  $ faucet [options]
+
+Options:
+  -h, --help
+    display this help message
+  -c, --config
+    configuration file (defaults to ${repr("faucet.config.js")})
+  -w, --watch
+    monitor the file system for changes to recompile automatically
+  --fingerprint
+    add unique hash to file names
+  --compact
+    reduce output size (e.g. via minification)
+`.trim();
+
+module.exports = function parseCLI(argv = process.argv.slice(2), help = HELP) {
```

Comment:
* Why give that function a name? 🤔 
* I would pass in the argv and not default to it
* Why not just use the "constant" HELP?

Reply:
* For debugging purposes, I've started paying more attention to functions being named¹ - not sure whether that's just another quixotic effort...
* Having to remember `.slice(2)` is a pita, so I included it.
* Without that `help` argument, alternative wrapper scripts wouldn't be able to reuse this function.

¹ you might recall this comparison from a while back:
```javascript
let foo = function() {};
```
```javascript
exports.foo = function() {};
```
```javascript
module.exports = {
    foo: function() {}
};
```

Reply:
Ok 😉